test/trim_trailing_whitespaces_test.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: true
+
+require_relative 'lib'
+
+def trim_trailing_whitespaces_round_trip(builder_block, expected, whitespace)
+  printer = Oppen::Wadler.new(width: 5, whitespace:)
+  builder_block.call printer
+  _(printer.output).must_equal expected
+
+  printer_no_trailing = Oppen::Wadler.new(width: 5, whitespace:)
+  builder_block.call printer_no_trailing, false
+  _(printer_no_trailing.output).must_equal expected
+end
+
+describe 'Trim trailing whitespaces tests' do
+  [
+    ' ',
+    '#',
+    '**',
+    ' *',
+    '* ',
+    ' * ',
+    '¬',
+    'éÅ',
+  ].each do |whitespace|
+    it "does not trim non trailing `#{whitespace}` whitespaces" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text("#{whitespace}a#{whitespace}")
+      printer.text("#{whitespace}b")
+      _(printer.output).must_equal "#{whitespace}a#{whitespace}#{whitespace}b"
+    end
+
+    it "trims a pure `#{whitespace}` whitespace token" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text(whitespace)
+      printer.break
+      _(printer.output).must_equal "\n"
+    end
+
+    it "trims multiple pure `#{whitespace}` whitespace tokens" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text(whitespace * 42)
+      printer.break
+      _(printer.output).must_equal "\n"
+    end
+
+    it "trims a single `#{whitespace}` trailing whitespace" do
+      builder_block = proc { |out, add_trailing = true|
```

Comment:
Avoid blocks, use lambdas (unless you need `break`, `return`, etc.).

```ruby
->(out, add_trailing = true) {
```

And chain calls instead of several `out.`.

Reply:
We cannot chain `out.` calls since all our methods (`text`, `break`, ...) do not return `self`. We can add this feature in the future if you want.